File: app/src/main/java/opencrowd/hgc/hgcwallet/ui/onboard/TermsScreen.java
package opencrowd.hgc.hgcwallet.ui.onboard;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.wealthfront.magellan.BaseScreenView;
import com.wealthfront.magellan.Screen;

import opencrowd.hgc.hgcwallet.R;
import opencrowd.hgc.hgcwallet.modals.HGCKeyType;
import opencrowd.hgc.hgcwallet.ui.customviews.TitleBarWrapper;

public class TermsScreen extends Screen<TermsView> {

    public boolean isAcceptBtnShow;
    public String content, title;

    public TermsScreen(boolean isAcceptBtnShow, String content, String title) {
        this.isAcceptBtnShow = isAcceptBtnShow;
        this.content = content;
        this.title = title;
    }

    @NonNull
    @Override
    protected TermsView createView(@NonNull Context context) {

        return new TermsView(context,isAcceptBtnShow, content, title);
    }
}


class TermsView extends BaseScreenView<TermsScreen> {
    Button mTermAccept;

    public TermsView(@NonNull Context context,boolean isAcceptBtnShow, String content, String title) {
        super(context);
        inflate(context, R.layout.fragment_term_layout,this);
        new TitleBarWrapper(findViewById(R.id.titleBar)).setTitle(title);
        mTermAccept = (Button)findViewById(R.id.btn_term_accept);
        TextView textView = findViewById(R.id.textView);
        textView.setText(content);

        if(isAcceptBtnShow == true) {
            mTermAccept.setVisibility(VISIBLE);
        } else {
            mTermAccept.setVisibility(GONE);
        }

        mTermAccept.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getScreen().getNavigator().goTo(new NewWalletCreateScreen(HGCKeyType.ED25519));
            }
        });
    }

}

File: app/src/main/java/hedera/hgc/hgcwallet/ui/onboard/TermsScreen.kt
package hedera.hgc.hgcwallet.ui.onboard

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.view.View
import android.widget.Button
import android.widget.TextView

import com.wealthfront.magellan.BaseScreenView
import com.wealthfront.magellan.Screen

import hedera.hgc.hgcwallet.Config
import hedera.hgc.hgcwallet.R
import hedera.hgc.hgcwallet.ui.customviews.TitleBarWrapper

class TermsScreen(showAcceptBtn: Boolean, content: String, title: String) : Screen<TermsView>() {

    data class Params(val showAcceptBtn: Boolean, val content: String, val title: String)

    private val params = Params(showAcceptBtn, content, title)
    override fun createView(context: Context): TermsView {
        return TermsView(context, params)
    }

    internal fun onAcceptClick() {
        navigator?.goTo(NewWalletCreateScreen())
    }

    internal fun viewFullTerms() {
        val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(Config.termsAndConditions))
        activity?.startActivity(browserIntent)
    }
}


class TermsView(context: Context, val params: TermsScreen.Params) : BaseScreenView<TermsScreen>(context) {

    init {
        View.inflate(context, R.layout.fragment_term_layout, this)
        TitleBarWrapper(findViewById(R.id.titleBar)).setTitle(params.title)

        findViewById<TextView>(R.id.textView)?.apply {
            text = params.content
        }

        findViewById<Button>(R.id.btn_term_accept)?.apply {
            visibility = if (params.showAcceptBtn) View.VISIBLE else View.GONE
            setOnClickListener { screen?.onAcceptClick() }
        }

        findViewById<Button>(R.id.btn_full_terms)?.apply {
            setOnClickListener { screen?.viewFullTerms() }
        }
    }

}